### PR TITLE
Improve crm_resource --constraints

### PR DESCRIPTION
**Pull request by kgaillot**
I did this to help investigate issues with the scheduler colocation code, but the rest of that project is not ready, so I figured I'd get this out in time for rc2. Normally I wouldn't put features in rc2, but this change has very limited scope and is well regression-tested, and if the rest of the project makes a later release candidate, this will be useful for testing it.

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , can you review this? Thanks!

File: tools/crm_resource.c [hunk 1]
```diff
@@ -396,12 +395,13 @@ static GOptionEntry query_entries[] = {
...
-    { "stack", 'A', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, flag_cb,
-      "Display the (co)location constraints that apply to a resource\n"
-      INDENT "and the resources is it colocated with",
-      NULL },
     { "constraints", 'a', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, flag_cb,
-      "Display the (co)location constraints that apply to a resource",
+      "Display the location and colocation constraints that apply to a\n"
+      INDENT "resource, and if --recursive is specified, to the resources\n"
+      INDENT "directly or indirectly involved in those colocations",
+      NULL },
```
Comment (nrwahl2) [inline tools/crm_resource.c:405]: This goes recursively in both directions, right?

File: tools/crm_resource.c [hunk 2]
```diff
@@ -1898,7 +1901,7 @@ main(int argc, char **argv)
 
         case cmd_colocations:
             rc = out->message(out, "locations-and-colocations", rsc, data_set,
-                              options.recursive);
+                              options.recursive, (bool) options.force);
```
Comment (nrwahl2) [inline tools/crm_resource.c:1904]: Wouldn't hurt to cast both if we cast one, but no need to re-push over that.

**Review by nrwahl2 — APPROVED** [review]
Looks good. You could add a test for the new `--constraints --recursive` option, but we're already testing the alias (`-A`), so it's not important.

File: tools/crm_resource.c (hunk 1 above)
Comment (kgaillot) [reply tools/crm_resource.c:405]: yep

**Pull request merged**